ncm-download/src/main/perl/download.pm

```diff
@@ -257,64 +225,48 @@ sub download {
         push @cmd, qw(--negotiate -u x:x);
     }
 
-    if ($key) {
-        push @cmd, ("--key", $key);
-    }
 
-    if ($cacert) {
-        push @cmd, ("--cacert", $cacert);
-    }
-
-    if ($capath) {
-        push @cmd, ("--capath", $capath);
-    }
-
-    if ($cert) {
-        push @cmd, ("--cert", $cert);
-    }
+    push(@cmd, map {("--$_", $opts{$_})} grep {$opts{$_}} qw(key cacert capath cert));
```

Comment:
The old code was too verbose but this is too far the other way. How about this as a half way house:
```
foreach my $key (qw(key cacert capath cert)) {
   push(@cmd, "--$key ".$opts{$key});
}
```